lib/src/value/calculation.dart

```diff
@@ -414,3 +868,14 @@ class CalculationInterpolation {
 
   String toString() => value;
 }
+
+//
+/// Extension methods to get the sign of the double's numerical value,
+/// including positive and negative zero.
+extension on double {
+  double get signIncludingZero {
+    if (identical(this, -0.0)) return -1.0;
+    if (this == 0) return 1.0;
+    return sign;
+  }
+}
```

Comment:
Extension methods should go in the `util/` directory. This should probably go in `util/number.dart`.